web/src/scheduled_messages_overlay_ui.js

```diff
@@ -56,29 +48,32 @@ export function launch() {
             browser_history.exit_overlay();
         },
     });
-    loading.make_indicator($("#scheduled_messages_overlay .loading-indicator"), {
-        abs_positioned: true,
+
+    const rendered_list = render_scheduled_message({
+        scheduled_messages_data: format(scheduled_messages.scheduled_messages_data),
     });
+    const $messages_list = $("#scheduled_messages_overlay .overlay-messages-list");
+    $messages_list.append(rendered_list);
+}
 
-    channel.get({
-        url: "/json/scheduled_messages",
-        success(data) {
-            hide_loading_indicator();
-            // Saving formatted data is helpful when user is trying to edit a scheduled message.
-            scheduled_messages.override_scheduled_messages_data(
-                format(data.scheduled_messages),
-            );
-            const rendered_list = render_scheduled_message({
-                scheduled_messages_data: scheduled_messages.scheduled_messages_data,
-            });
-            const $messages_list = $("#scheduled_messages_overlay .overlay-messages-list");
-            $messages_list.append(rendered_list);
-        },
-        error(xhr) {
-            hide_loading_indicator();
-            blueslip.error(xhr);
-        },
+export function rerender() {
+    if (!overlays.scheduled_messages_open()) {
+        return;
+    }
+    const rendered_list = render_scheduled_message({
+        scheduled_messages_data: format(scheduled_messages.scheduled_messages_data),
     });
+    const $messages_list = $("#scheduled_messages_overlay .overlay-messages-list");
+    $messages_list.find(".scheduled-message-row").remove();
+    $messages_list.append(rendered_list);
+}
+
+export function remove_scheduled_message_id(scheduled_msg_id) {
+    if (overlays.scheduled_messages_open()) {
+        $(
+            `#scheduled_messages_overlay .scheduled-message-row[data-message-id=${scheduled_msg_id}]`,
```

Comment:
This can be a follow-up, but we should definitely call it `data-scheduled-message-id`, to avoid confusion -- they are not message IDs.